src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -1031,6 +1028,1438 @@ static void ReportField(FieldInfo field)
+            }
+            if (missingConstructor)
+            {
+                comp.MakeMemberMissing(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor);
+            }
+            string expectedAttributes = (missingType || missingConstructor) ? "" : " System.Runtime.CompilerServices.CompilerGeneratedAttribute,";
+            CompileAndVerify(comp, verify: Verification.Skipped, expectedOutput: IncludeExpectedOutput($$"""
+                C.<P1>k__BackingField:{{expectedAttributes}}
+                C.<P2>k__BackingField:{{expectedAttributes}}
+                C.<P3>k__BackingField:{{expectedAttributes}}
+                C.<P4>k__BackingField:{{expectedAttributes}}
+                """));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void Conditional(bool useDEBUG)
+        {
+            string sourceA = """
+                using System.Diagnostics;
+                class C
+                {
+                    public static object P1 { get { M(field); return null; } set { } }
+                    public static object P2 { get { return null; } set { M(field); } }
+                    public object P3 { get { M(field); return null; } }
+                    public object P4 { set { M(field); } }
+                    public object P5 { init { M(field); } }
+                    [Conditional("DEBUG")]
+                    static void M( object o) { }
+                }
+                """;
+            string sourceB = """
+                using System;
+                using System.Reflection;
+                class Program
+                {
+                    static void Main()
+                    {
+                        foreach (var field in typeof(C).GetFields(BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static))
+                            ReportField(field);
+                    }
+                    static void ReportField(FieldInfo field)
+                    {
+                        Console.Write("{0}.{1}:", field.DeclaringType.Name, field.Name);
+                        foreach (var obj in field.GetCustomAttributes())
+                            Console.Write(" {0},", obj.ToString());
+                        Console.WriteLine();
+                    }
+                }
+                """;
+            var parseOptions = TestOptions.RegularNext;
+            if (useDEBUG)
+            {
+                parseOptions = parseOptions.WithPreprocessorSymbols("DEBUG");
+            }
+            var verifier = CompileAndVerify(
+                [sourceA, sourceB],
+                parseOptions: parseOptions,
+                targetFramework: TargetFramework.Net80,
+                verify: Verification.Skipped,
+                expectedOutput: IncludeExpectedOutput("""
+                    C.<P3>k__BackingField: System.Runtime.CompilerServices.CompilerGeneratedAttribute,
```

Comment:
nit: feels like it would be good to verify presence of the field symbols in NamedTypeSymbol.GetMembers() as well.